### ISSUE TITLE
[EXPORT] 

### ISSUE THREAD

**Issue by EstebanFPV** [olive-editor/olive#2211]
**Commit Hash** 
5626e6ae

**Platform** 
Kubuntu 22.10

**Summary**
Whenever i try to export a video olive crashes as soon as i click export. Started happening with the previous nightly build so i updated to the 3/30/23 one and the issue persists. 
Returns: SIGSEGV /SEGV_MAPERR
I haven't found anything else in the log that looks useful

**Additional Information / Output**

**Comment by itsmattkc** [comment]
This doesn't really tell me much. Exports are fine on my end. Do you get a crash report? If it started in a recent commit, can you find the last commit that didn't crash?

**Comment by ThomasWilshaw** [comment]
Can you share the crash report you're getting?

**Comment by EstebanFPV** [comment]
> Can you share the crash report you're getting?

how do i do that?

on another note. i managed to get that video to export, seemed to be something to do with my timeline settings?
now i have another video that says it renders ~22-27 seconds of video then quits counting. the only way to exit the exporting dialogue is to click the x button, which crashes the program.

**Comment by itsmattkc** [comment]
> how do i do that?

In your video, there's a crash report that appears that you then click "Don't Send" on. Instead, copy and paste the contents here.

**Comment by EstebanFPV** [comment]
> > how do i do that?
> 
> In your video, there's a crash report that appears that you then click "Don't Send" on. Instead, copy and paste the contents here.

With my latest issue no crash report comes up

**Comment by itsmattkc** [comment]
Then use GDB: https://github.com/olive-editor/olive/wiki/Debugging-Olive

**Comment by itsmattkc** [comment]
Are you absolutely sure it doesn't finish exporting if you were to leave it without cancelling?

**Comment by EstebanFPV** [comment]
> Are you absolutely sure it doesn't finish exporting if you were to leave it without cancelling?

i mean, i left it alone for a half hr

**Comment by EstebanFPV** [comment]
![image](https://user-images.githubusercontent.com/71269805/230739315-c398989b-49c2-4e8c-b07d-16ee3f262f4a.png)

**Comment by itsmattkc** [comment]
When is this? Is this during export?

**Comment by EstebanFPV** [comment]
> When is this? Is this during export?

Yes, i opened olive, tried to export, and then force closed it when it quit at around 26 seconds

**Comment by itsmattkc** [comment]
There's some kind of issue happening with one of the footage files you're using. It's trying to read the audio and encountering an error, so it tries again, encounters the same error, and does this infinitely, hence why the export hangs. Ideally it would cancel the export the first time it hits this error, but I've never encountered a file that does this, so it's never been written to handle it.

Would you be able to send this file? The log doesn't say what file exactly, but it's one with AAC audio in it.

**Comment by EstebanFPV** [comment]
> There's some kind of issue happening with one of the footage files you're using. It's trying to read the audio and encountering an error, so it tries again, encounters the same error, and does this infinitely, hence why the export hangs. Ideally it would cancel the export the first time it hits this error, but I've never encountered a file that does this, so it's never been written to handle it.
> 
> Would you be able to send this file? The log doesn't say what file exactly, but it's one with AAC audio in it.

there's a lot of video files in this project. i did have olive automatically select timeline settings i think off of a 2.7k 30 gopro file, and when i go to export it is exporting audio as AAC. if i switch the audio to mp3 the issue persists tho

**Comment by EstebanFPV** [comment]
all files are .mp4 from a phone and gopro, with a couple .mp3 music files

**Comment by itsmattkc** [comment]
It's not about exporting to AAC or Olive's sequence settings, it's about one of the files that happens to contain AAC audio. It may be slightly corrupt, which is what's tripping up FFmpeg/Olive and throwing the error.

The error in question is that `channel element 2.1 is not allocated` message. Maybe try scrubbing the timeline with a log/terminal side by side and see which clip/file throws that error (make sure audio scrubbing is on because it's definitely an audio error).

**Comment by EstebanFPV** [comment]
and it could be any file, not nescesarily the one at 26 seconds?

**Comment by itsmattkc** [comment]
No, the timers in the export dialog count realtime (i.e. how many real hours/minutes/seconds have elapsed), not how much of the sequence has been rendered.

**Comment by EstebanFPV** [comment]
![image](https://user-images.githubusercontent.com/71269805/230740476-f964dc1f-3e27-4a91-ae56-b32b5653561a.png)

**Comment by EstebanFPV** [comment]
didn't touch the timeline (just loaded the proj) and got all this

**Comment by itsmattkc** [comment]
Yes, it will also try to access the audio files on startup to generate the timeline waveforms, and is again encountering the same error. But it still doesn't tell us which file(s) are causing the problem.

**Comment by EstebanFPV** [comment]
now i have no audio playback in olive

**Comment by EstebanFPV** [comment]
found an audio track that throws the 2.1 thing, we'll see what happens without that

**Comment by EstebanFPV** [comment]
![image](https://user-images.githubusercontent.com/71269805/230740649-9d624b92-bf83-4208-bc7a-9396b48678fa.png)
 nope...

**Comment by itsmattkc** [comment]
If it's still throwing them, then there are more with that same issue.

**Comment by itsmattkc** [comment]
If you can send me one of the files that triggers that error, I can improve the way Olive handles it.

**Comment by EstebanFPV** [comment]
> If you can send me one of the files that triggers that error, I can improve the way Olive handles it.

I'll make sure it's the files i think it is that's actually causing the issue quick

**Comment by EstebanFPV** [comment]
Now it just straight up crashed, i didn't even have to close it... All i did was get rid of 2 audio clips, and save

**Comment by EstebanFPV** [comment]
I scrubbed through it and couldn't find any more, though it did say "failed to pull from buffersink" quite a bit

**Comment by itsmattkc** [comment]
> I scrubbed through it and couldn't find any more, though it did say "failed to pull from buffersink" quite a bit

That's not a good sign either. If you can send a file that causes this error too, that would be helpful.

**Comment by EstebanFPV** [comment]
well, that will be more difficult, quite a few files returned that error

**Comment by itsmattkc** [comment]
I only need one.

**Comment by EstebanFPV** [comment]
seems like it's mostly nested sequences for both errors

**Comment by EstebanFPV** [comment]
[In the Sweet By and By - Zachariah Hickman.zip](https://github.com/olive-editor/olive/files/11184193/In.the.Sweet.By.and.By.-.Zachariah.Hickman.zip)

**Comment by EstebanFPV** [comment]
there's one that threw the 2.1 error i think

**Comment by EstebanFPV** [comment]
it doesn't seem to like any of my nested sequences of gopro videos that are sped up in olive.

**Comment by itsmattkc** [comment]
It can't be this one. The error is during an AAC read and this is an MP3 file.

**Comment by EstebanFPV** [comment]
the 2.1 error went away from that part of the timeline when i removed that file :shrug:

**Comment by EstebanFPV** [comment]
i'd send you my project file and all my assets, but that would be a rather large .zip file

**Comment by itsmattkc** [comment]
Often when I can't reproduce issues on my end, I offer to remote in so I can see exactly what's happening myself and try to fix it. Would you be open to doing that? If so, I'll add you on Discord and we can go from there.

**Comment by EstebanFPV** [comment]
ok

**Comment by EstebanFPV** [comment]
i'm on linux if that makes it harder or easier. 
why, i really have no idea. i'm not smart enough to use it properly.
